File: src/main/Java/others/boj15829.java
import java.io.*;
import java.math.BigInteger;

public class boj15829 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        char[] line = reader.readLine().toCharArray();
        BigInteger sum = BigInteger.valueOf(0);
        for (int i = 0; i < n; i++) {
            int tmp = line[i] - 'a' + 1;
            BigInteger base = BigInteger.valueOf(31);
            BigInteger exponent = base.pow(i);
            BigInteger tmpBigInt = BigInteger.valueOf(tmp);
            sum = sum.add(tmpBigInt.multiply(exponent));
        }
        writer.append(sum.mod(BigInteger.valueOf(1234567891)).toString()).close();
    }
}

File: src/main/kotlin/others/boj15829.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.math.BigInteger

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    r.readLine()
    val line = r.readLine()
    var sum = BigInteger.valueOf(0)
    line.forEachIndexed { index, it ->
        val tmp = it - 'a' + 1
        val under = BigInteger.valueOf(31)
        val tmpBigInt = BigInteger.valueOf(tmp.toLong())
        sum = sum.add(tmpBigInt * under.pow(index))
    }
    w.append("${sum.mod(BigInteger.valueOf(1234567891))}").close()
}